Add `drop_metrics` and `drop_tags` to cloud options (#1018)
grafana/k6-docs: src/data/markdown/docs/03 cloud/01 Creating and running a test/02 Cloud tests from the CLI/200 Cloud execution reference.md

| note (string)                 | Empty by default.                                                                      | Notes regarding the test, changes made, or anything that may be worth noting about your test.                                                                                                                         |
| deleteSensitiveData (boolean) | False by default                                                                       | If set to `true`, k6 deletes sensitive data as soon as the test starts running or, if still queued, when the test aborts. Sensitive data includes scripts, HAR files, archives, and APM credentials.                  |
| drop_metrics (array)          | Empty by default                                                                       | Drops listed metrics during ingestion time. E.g. `["http_req_tls_handshaking"]`. This helps reduce the cardinality of time series.                                                                                    |

> | drop_tags (object)            | Empty by default                                                                       | Drops listed tags during ingestion time. E.g. `{"http_req_duration": ["instance_id"]}`. This helps reduce the cardinality of time series.                                                                             |

Reviewer: >Drops listed tags during ingestion t

I probably won't review this until today, but for a quick preliminary question: what are "Listed metrics"? The metrics calculated? Or the ones stored and shown in the UI? It seems like it calculates, but doesn't store in the DB (and therefore isn't shown of course).

Reviewer: Or maybe are they metrics listed in an array in the script somewhere?

Author: By listed I meant the ones in the array. As you didn't understand, I can propose:

```suggestion
Drops the metrics listed in the array during ingestion time.
```

Author: By metrics, I mean metrics from this list: https://k6.io/docs/using-k6/metrics/#http-specific-built-in-metrics

> | drop_metrics (array)          | Empty by default                                                                       | Drops listed metrics during ingestion time. E.g. `["http_req_tls_handshaking"]`. This helps reduce the cardinality of time series.                                                                                    |
> | drop_tags (object)            | Empty by default                                                                       | Drops listed tags during ingestion time. E.g. `{"http_req_duration": ["instance_id"]}`. This helps reduce the cardinality of time series.                                                                             |

Reviewer: ```suggestion
| drop_metrics (array)          | Empty by default                                                                       | Drops the metrics listed in the array during ingestion time.
 E.g. `["http_req_tls_handshaking"]`. This helps reduce the cardinality of time series.                                                                                    |
| drop_tags (object)            | Empty by default                                                                       | Drops tags for a specified metric, where the metric is the key and the tags are an array. E.g. `{"http_req_duration": ["instance_id"]}`. This helps reduce the cardinality of time series.                                                                             |
```

</Blockquote>

> <Blockquote mod="note" title="">

Reviewer: ```suggestion
### Options to reduce time series
```

> The `drop_metrics` and `drop_tags` options are for advanced users that are following good practices
> when writing their tests but are still hitting the "Too many time series" alert, resulting in an

Reviewer: ```suggestion
The `drop_metrics` and `drop_tags` options are for advanced users that are following good practices
when writing their tests but are still hitting the "Too many time series" alert
```

I don't think we should say advanced or "good practices" since these are subjective terms.

Can you provide an example of when a user might use these options and why they might benefit? Like what's a good reason to generate so many metrics, and what benefit comes from dropping them? I think we can reframe this as an example of a rare case for certain circumstances, rather than ask readers to decide whether they're advanced or not.

Author: We had some users that even following these tips, still hit the limits:

<img width="668" alt="Screenshot 2023-01-31 at 15 25 31" src="https://user-images.githubusercontent.com/23944/215787033-dc2718ee-bd43-4ae3-9103-1ae03872fa70.png">

What the screenshot explains is what I meant by "good practices".

Reviewer 2: @rafa-munoz it might be good to add this info to the Performance Insights tooltip in the app as well if we haven't already.

Reviewer: I think for now, we can at least link this from the Performance insights page  (and back)

Reviewer: Also it appears we need to document the "Too many time series" performance metric :laughing:

Author: @Vmihajlovic I will add a mention back in the tooltip when we have this merged

> The `drop_metrics` and `drop_tags` options are for advanced users that are following good practices
> when writing their tests but are still hitting the "Too many time series" alert, resulting in an

Reviewer: ```suggestion
The `Too many time series` alert aborts a test run.
Sometimes, though, a test might trigger the alert even if they follow all the recommended practices from the performance insight.
In these cases, you can consider the `drop_metrics` and `drop_tags` options.
```

aborted test run.

> Please be aware that its usage may have implications on showing graphs, which may appear empty for

Reviewer: ```suggestion
Note that if you use these options, some graphs may appear empty for
```

> aborted test run.
> 
> Please be aware that its usage may have implications on showing graphs, which may appear empty for
> the metrics deleted.
> 
> Because they are core part of a test, some metrics cannot be dropped, as `http_reqs` or
> `http_req_duration`. Neither tags `name` and `method` from `http_req_duration`.
> 
> </Blockquote>

Reviewer: ```suggestion
aborted test run.

Please be aware that its usage may have implications on showing graphs, which may appear empty for
the metrics deleted.

Certain important tags and metrics cannot be dropped:
- For metrics,  you can't drop `http_reqs` or `http_req_duration`.
- For tags, you cannot drop `name` and `method` from `http_req_duration`.

```

</Blockquote>

### Options to reduce time series

The `Too many time series` alert aborts a test run.

> Sometimes, though, a test might trigger the alert even if they follow all the recommended practices from the performance insight.

Reviewer: ```suggestion
Sometimes, though, a test might trigger the alert even if the script follows all recommended practices from the performance insight.
```

In these cases, you can consider the `drop_metrics` and `drop_tags` options.

> aborted test run.

Reviewer: ```suggestion
```

Author: Oops, I didn't realise about this

Note that if you use these options, some graphs may appear empty for
the metrics deleted.

Certain important tags and metrics cannot be dropped:
- For metrics,  you can't drop `http_reqs` or `http_req_duration`.

> - For tags, you cannot drop `name` and `method` from `http_req_duration`.

Reviewer: ```suggestion
- For tags, you can't drop `name` and `method` from `http_req_duration`.
```
